### PR TITLE
Add a unit test to require registry.npm.js as the package registry.

### LINKED ISSUE #612: Enforce only public NPM registries in package-lock.json
A non-public NPM repository snuck into our config accidentally which broke things.

We should add a unit test that scans package-lock.json and requires npm as the registry.

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #612

### CODE REVIEW

File: src/package_test.ts [hunk 1]
```diff
@@ -7,4 +7,13 @@ describe('package', () => {
         const v2 = require('../package-lock.json').version;
         expect(v1).to.equal(v2);
     });
+
+    it('package-lock should only reference npm', () => {
+        const deps = require('../package-lock.json').dependencies;
+        for (const key in deps) {
+            const dep = deps[key];
+            const resolved = new URL(dep.resolved);
+            expect(resolved.hostname).to.equal('registry.npmjs.org');
```
Comment (drubin) [inline src/package_test.ts:16]: Can we also add a HTTPS check here too while we at it. 

```suggestion
            expect(resolved.hostname).to.equal('registry.npmjs.org');
            expect(resolved.protocol ).to.equal('https);
```

Its too easy to mistakenly switch to the HTTP registry and I doubt either you or I would notice it.

File: src/package_test.ts [hunk 2]
```diff
@@ -7,4 +7,13 @@ describe('package', () => {
         const v2 = require('../package-lock.json').version;
         expect(v1).to.equal(v2);
     });
+
+    it('package-lock should only reference npm', () => {
```
Comment (drubin) [inline src/package_test.ts:11]:
```suggestion
    it('package-lock should only reference official public npm registry', () => {
```

lets be more exact and specific here too.

**Review by drubin — CHANGES_REQUESTED** [review]
Minor suggestion, but it looks good. if you make the changes I will approve.

Also thanks for this! Its surprisingly clean

**Comment by brendandburns** [comment]
Comments addressed.

File: src/package_test.ts [hunk 3]
```diff
@@ -7,4 +7,14 @@ describe('package', () => {
...
         expect(v1).to.equal(v2);
     });
+
+    it('package-lock should only reference official public npm registry', () => {
+        const deps = require('../package-lock.json').dependencies;
+        for (const key in deps) {
+            const dep = deps[key];
+            const resolved = new URL(dep.resolved);
+            expect(resolved.hostname).to.equal('registry.npmjs.org');
+            expect(resolved.protocol ).to.equal('https);
```
Comment (drubin) [inline src/package_test.ts:17]:
```suggestion
            expect(resolved.protocol).to.equal('https);
```

not sure how this space got in here.

File: src/package_test.ts [hunk 4]
```diff
@@ -7,4 +7,14 @@ describe('package', () => {
         const v2 = require('../package-lock.json').version;
         expect(v1).to.equal(v2);
     });
+
+    it('package-lock should only reference official public npm registry', () => {
+        const deps = require('../package-lock.json').dependencies;
+        for (const key in deps) {
```
Comment (dominykas) [inline src/package_test.ts:13]: It's not enough to loop through the top level - each of `dependencies` can have its own `dependencies`, e.g. https://github.com/kubernetes-client/javascript/blob/d18e69d531580db628a4e925f0ece42a80a20412/package-lock.json#L38

File: src/package_test.ts (hunk 3 above)
Comment (brendandburns) [reply src/package_test.ts]: fixed.

File: src/package_test.ts (hunk 4 above)
Comment (brendandburns) [reply src/package_test.ts]: done.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!

**Comment by brendandburns** [comment]
@drubin @dominykas 

Friendly ping on this one.

Thanks!

**Review by dominykas — APPROVED** [review]

**Comment by drubin** [comment]
/lgtm 

Thanks!

**Pull request merged**